### PR TITLE
Add application versioning configuration

### PR DESCRIPTION
**Pull request by janybravo**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -30,6 +30,8 @@ jobs:
       - name: Install NPM packages
         run: npm --cache-min 86400 install
       - name: Build staging bundle
+        env:
+          GENERATE_SOURCEMAP: false
```
Comment (dblenkus) [inline .github/workflows/build.yml]: We need the source map. Otherwise, it will be hard to debug, and also Sentry depends on it.

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,7 +3,7 @@
 All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 
-## [Unreleased]
+## 1.0.0 [Unreleased]
```
Comment (dblenkus) [inline CHANGELOG.md]: I would keep just "Unreleased" until it is released. The main reason is that you don't know which version will be next.

File: README.md [hunk 3]
```diff
@@ -31,11 +31,9 @@ Typescript compiler, Prettier and ESLint checks combined.
 Launches the test runner in the interactive watch mode.<br />
 See the section about [running tests](https://facebook.github.io/create-react-app/docs/running-tests) for more information.
 
-### `npm run test:e2e`
+### `npm run e2e:debug`
```
Comment (dblenkus) [inline README.md:34]: These changes are not related to versioning, so they should go into a separate commit.

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "dicty-express",
-    "version": "0.1.0",
+    "version": "1.0.0",
```
Comment (dblenkus) [inline package.json]: Set this to `1.0.0a1` until it is actually released.

File: .github/workflows/build.yml (hunk 1 above)
Comment (janybravo) [reply .github/workflows/build.yml]: This means source code for others via sourcemaps, if that's ok with us. I'm used to deal with packages without source maps, but its easier with them anyway. You suggest we keep them.

File: README.md (hunk 3 above)
Comment (janybravo) [reply README.md:34]: done

File: CHANGELOG.md (hunk 2 above)
Comment (janybravo) [reply CHANGELOG.md]: This way we can keep version values simple (eg. no postfix). Unreleased tells us its not ready for others, also tells us what is to be next for them. We can always change unreleased to something else according to business needs. I think this way is better than to have aditional postfix, lets discuss.

Comment (janybravo) [reply CHANGELOG.md]: Ok, its convinience to our workflow to have no exact version with "Unreleased" keyword in changelog as we preffer relevant one at eg. package.json.

File: package.json (hunk 4 above)
Comment (janybravo) [reply package.json]: Ok, its convinience to our workflow to have an a posfix that equals unique git tag for deployment and possibly for debugging.

File: .github/workflows/build.yml (hunk 1 above)
Comment (janybravo) [reply .github/workflows/build.yml]: I see, they won't be accessible without special token anyways, should remain

File: .env.production [hunk 5]
```diff
@@ -0,0 +1,4 @@
+REACT_APP_VERSION=${npm_package_version}
```
Comment (dblenkus) [inline .env.production]: This is not yet a production version. Is there a way to differentiate between staging and production?

File: .env.production [hunk 6]
```diff
@@ -0,0 +1,4 @@
...
+REACT_APP_NAME=dictyExpress
+REACT_APP_SENTRY_URL=https://e0a554a693f641a6923a4240a2a64dfb@o74924.ingest.sentry.io/5265645
+REACT_APP_WEBSOCKET_URL=wss://qa2.genialis.com/ws/
```
Comment (dblenkus) [inline .env.production]: Add a new line at the end.

File: src/components/genexpress/genexpressAppBar/genexpressAppBar.tsx [hunk 7]
```diff
@@ -126,7 +127,8 @@ const GenexpressAppBar = ({
         <DesktopSectionContainer>
             <TitleContainer>
                 <DictyLogo src={dictyLogo} alt="dictyExpress logo" />
-                <GenexpressTitle>dictyExpress</GenexpressTitle>
+                <GenexpressTitle>{process.env.REACT_APP_NAME}</GenexpressTitle>
```
Comment (dblenkus) [inline src/components/genexpress/genexpressAppBar/genexpressAppBar.tsx:130]: Include `process.env.REACT_APP_COMMIT_SHA` for development versions.

**Review by dblenkus — COMMENTED** [review]
Looks good in general. Please squash commits before I approve.

File: .env.production (hunk 5 above)
Comment (janybravo) [reply .env.production]: environment variable describing environment will be defined in GHA build script per environment.

File: src/components/genexpress/genexpressAppBar/genexpressAppBar.tsx (hunk 7 above)
Comment (janybravo) [reply src/components/genexpress/genexpressAppBar/genexpressAppBar.tsx:130]: not here

File: .env.production (hunk 6 above)
Comment (janybravo) [reply .env.production]: added. but i would like to know why it is important to you besides githubs - icon?

File: .env.production (hunk 5 above)
Comment (dblenkus) [reply .env.production]: I know, but this should be a staging configuration, as it also contains Sentry configuration which is different for staging and production servers.

File: .env.production (hunk 6 above)
Comment (dblenkus) [reply .env.production]: There are 2 main reasons:
- it is easier to select a line in text editors if there is a line below
- adding a line after the last one changes two lines (newline character in the first line + the actual line)

File: src/components/genexpress/genexpressAppBar/genexpressAppBar.tsx (hunk 7 above)
Comment (dblenkus) [reply src/components/genexpress/genexpressAppBar/genexpressAppBar.tsx:130]: Where will you add it?

Comment (dblenkus) [reply src/components/genexpress/genexpressAppBar/genexpressAppBar.tsx:130]: Ok, i see.

File: .env [hunk 8]
```diff
@@ -1,6 +1,4 @@
-REACT_APP_VERSION=${npm_package_version}-development
+REACT_APP_VERSION=${npm_package_version}
 REACT_APP_NAME=dictyExpress
-REACT_APP_API_URL=http://localhost:3000/api
```
Comment (dblenkus) [inline .env:1]: The commit should use "environment variables" instead of "variables" to clarify what was removed. Also, you can name  "api/ and rest-api/" simply "API endpoint".

File: .env.staging [hunk 9]
```diff
@@ -1,7 +0,0 @@
-REACT_APP_VERSION=${npm_package_version}
```
Comment (dblenkus) [inline .env.staging]: Why was the staging configuration removed?

Comment (janybravo) [reply .env.staging]: yes

File: .env.production (hunk 5 above)
Comment (janybravo) [reply .env.production]: `production` means a building type, not our target environment - that will be defined there as agreed with @acopar . For sentry address is easier to configure via env variable at github actions scripts. I think it should stay that way.

File: .env [hunk 10]
```diff
@@ -1,6 +1,2 @@
-REACT_APP_VERSION=${npm_package_version}-development
```
Comment (dblenkus) [inline .env]: Try to shorten the commit message, so it won't be cut on the Github.

**Review by dblenkus — APPROVED** [review]
